src/nix.rs

```diff
@@ -13,10 +13,11 @@ pub fn get() -> io::Result<OsString> {
     // host names are limited to `HOST_NAME_MAX` bytes
     //
     // https://pubs.opengroup.org/onlinepubs/9699919799/functions/gethostname.html
-    let size =
-        unsafe { libc::sysconf(libc::_SC_HOST_NAME_MAX) as libc::size_t };
+    let limit =
+        unsafe { libc::sysconf(libc::_SC_HOST_NAME_MAX) };
+    let size = if limit < 0 { 255 } else { limit } as usize;
```

Comment:
I'm assuming that this change goes from the following statement?

> Max length of a hostname, not including the terminating null byte, as returned by `gethostname(2)`. Must not be less than `_POSIX_HOST_NAME_MAX` (255). 

It seems that `_POSIX_HOST_NAME_MAX` is [not defined](https://docs.rs/libc/latest/libc/?search=_POSIX_HOST_NAME_MAX) in `libc` crate, so that's why it is hardcoded as a simple `255` here?

Reply:
That's correct. Unfortunately I did not find an already defined constant anywhere.

Reply:
Indeed, it is not defined anywhere for some (?) reasons. Do you think it would be better define it here locally for better readability and understanding of what is happening exactly?

Ex. 
```rust
const _POSIX_HOST_NAME_MAX = 255;
```

I'm also looking at [the documentation here](https://man7.org/linux/man-pages/man3/sysconf.3.html)

>         `HOST_NAME_MAX` - `_SC_HOST_NAME_MAX`
>              Maximum length of a hostname, not including the
>              terminating null byte, as returned by `gethostname(2)`.
>              Must not be less than `_POSIX_HOST_NAME_MAX` (255).

Perhaps we should do smth like `max(limit, _POSIX_HOST_NAME_MAX)` instead to fully conform the specification?
Please keep in mind that it is just a suggestion, discussions are more that welcomed :)

Reply:
I thought about something like that as well. We could still run into issues if sysconf returns a huge value (larger than stack), but on the other hand the upper limit seems to be kind of fixed. Maybe a constant "255" without even asking sysconf is even easier...

I have adjusted the pull request. Let me know if it fits your style requirements and ideas.